tarogarden/caretaker.go

```diff
@@ -322,23 +322,42 @@ func (b *BatchCaretaker) seedlingsToAssetSprouts(ctx context.Context,
 		}
 
 		var groupKey *asset.GroupKey
-		// If emission is enabled, then we'll need to generate another
-		// public key, then use that to derive the key group signature
-		// along with the tweaked key group.
 		if seedling.EnableEmission {
-			rawGroupKey, err := b.cfg.KeyRing.DeriveNextKey(
-				ctx, asset.TaroKeyFamily,
-			)
-			if err != nil {
-				return nil, fmt.Errorf("unable to derive "+
-					"group key: %v", err)
-			}
-			groupKey, err = asset.DeriveGroupKey(
-				b.cfg.GenSigner, rawGroupKey, assetGen, nil,
-			)
-			if err != nil {
-				return nil, fmt.Errorf("unable to tweak	group "+
-					"key: %v", err)
+			switch seedling.HasGroupKey() {
+			// If the seedling has a group key specified,
+			// that group key was validated earlier. We need to
+			// sign the new genesis with that group key.
+			case true:
+				groupKey, err = asset.DeriveGroupKey(
+					b.cfg.GenSigner,
+					seedling.GroupInfo.GroupKey.RawKey,
+					*seedling.GroupInfo.Genesis, &assetGen,
+				)
+				if err != nil {
+					return nil, fmt.Errorf("unable to"+
+						"tweak group key: %v", err)
+				}
+
+			// If emission is enabled without a group key specified,
```

Comment:
Retracting my joke from above, since it actually does make more sense to use a `switch` statement here instead of `if/else`, as that allows you to add comments in a nicer way. I like :+1: